src/backend/distributed/planner/distributed_planner.c

```diff
@@ -242,14 +248,24 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 				result->planTree->total_cost = FLT_MAX / 100000000;
 			}
 		}
+
+		PlannerLevel = originalLevel;
```

Comment:
Other places [1] where we keep counters like this we perform a decrement both here and in the `PG_CATCH`. I have two questions:

1. Is the use of  `originalLevel` safe in `PG_CATCH`? I thought all local variables are unsafe to use during rollback, hence we were decrementing global variables in other places:
2. Is it unsafe to decrement? Would there be a possibility of missing a decrement?

[1] https://github.com/citusdata/citus/blob/2829c601dd48dcdbbbf3c748777f0b1b056484fa/src/backend/distributed/commands/utility_hook.c#L269